code/datums/components/riding/riding_mob.dm

```diff
@@ -277,6 +278,15 @@
 	else if(riding_offsets["[RIDING_OFFSET_ALL]"])
 		. = riding_offsets["[RIDING_OFFSET_ALL]"]
 
+/// If the crusher jumps, the rider(s) jumps alongside them
+/datum/component/riding/creature/crusher/proc/make_rider_jump()
+	SIGNAL_HANDLER
+
+	var/atom/movable/parentmovable = parent
+	for(var/mob/living/rider AS in parentmovable.buckled_mobs)
+		var/datum/component/jump/riderjump = rider.GetComponent(/datum/component/jump)
+		riderjump?.do_jump(rider, buckled_check = FALSE)
```

Comment:
this is stinky for several reasons. Its using getcomponent when it should be using signals. It being stinky in other spots isn't a reason to also do it.
Also this is not something specific to crusher riding. Anyone piggybacking is going to have this issue.
Like I said in discord, it needs to be split up so the rider just takes the jump args from the jumper instead of actually jumping themselves.

Reply:
resolved with rewrite